app/services/prompt_augmentor.rb

```diff
@@ -12,19 +12,30 @@ def augment
 
   def prompt
     @prompt ||= if @search_hits.any?
-      prompt = "Here is some context that may help you answer the following question:\n\n"
+      prompt = <<~CONTENT
+        You are given a query to answer based on some given textual context, all inside xml tags.
+        If the answer is not in the context but you think you know the answer, explain that to the user then answer with your own knowledge.
+
+      CONTENT
       @search_hits.each do |hit|
-        prompt << "#{hit.content}\n\n"
+        prompt << prompt_context(hit)
       end
-
-      prompt << "Question: #{@message.content}\n"
+      prompt << "<user_query>\n#{@message.content}\n<user_query>\n"
     else
       @message.content
     end
   end
 
   private
 
+  def prompt_context(hit)
+    if hit.document.web?
```

Comment:
In my local env I'm getting a NoMethodError from `.web?` here:

```
jobs    | NoMethodError: undefined method `web?' for nil:NilClass
jobs    |
jobs    |     if hit.document.web?
jobs    |                    ^^^^^/Users/nick/src/github.com/nickthecook/archyve-main/app/services/prompt_augmentor.rb:32:in `prompt_context'
```

Upon investigation, it looks like it's because sometimes a SearchHit is a Chunk, and has a `.document`, but sometimes a SearchHit is a GraphEntity and doesn't. In PromptAugmentor, you're going to have to handle both of these cases. In the future, there may be even more cases, like `hit.reference` being an instance of GraphRelationship.

Do you have any Collections with knowledge graph enabled?

Reply:
Oh. I see. It's a little confusing, but I fixed to check that `hit.document` is not nil.   Maybe there should be rspec tests for the different things that can be referenced? (which would have failed earlier)

Reply:
But now I'm not sure if that's correct.  If there's a hit on a GraphEntity, shouldn't I return something instead of blank context?
```ruby
return "" unless hit.document

Reply:
Yeah, you found augmentations.

If you want to feed the link to the LLM, put it in the prompt context. If you just want it available for the user, put the info in the augmentation.

There are partials for displaying augmentations differently depending on whether they're chunks or entities, so you could just update the partial for displaying chunks and add the extra info there.

You could also do one PR with just the new prompt format and the XML tags, and do the link stuff in a separate PR.

Reply:
I added a test for the search hit on an entity.  I'm sure this can be cleaned up, but maybe would be nice to try it and see how it performs?

Also, there's an error on test github step.

Reply:
So, what do you think? Can you just add the prompt change without the link info showing up in the UI for now, and add the augmentation stuff later? Seems like most of the complications in this one are around getting the link from the hit, whereas your prompt improvement is clear and simple.

Reply:
So the link gets added to the context now and I think everything works and the tests pass, but you want to wait to do that better later?

Reply:
Oh wait.  Sorry.  Now I see the error in `rspec ./spec/services/graph/summarize_collection_entities_spec.rb`.

I'm hoping it's minor and will fix it tonight or tomorrow.

Reply:
Fixed.  Sorry about that.

Reply:
No worries!

I was just thinking that maybe feeding the link to the LLM isn't going to help, but I'll have a look and can adjust later.